### PR TITLE
GEOT-6757 : Tile client throwing exception while downloading

### PR DESCRIPTION
**Pull request by roarbra**
Moved ImageLoading to TileService instead of Tile. And using HTTPClient to ensure we have an InputStream pointing to an image.

# Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [X] Confirm you have read the [contribution guidelines](https://github.com/geotools/geotools/blob/master/CONTRIBUTING.md)
- [X] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [X] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.
- [x] The changes are not breaking the build in downstream projects using SNAPSHOT dependencies, GeoWebCache and GeoServer (there is an automatic PR check verifying this, check this when it turns green).

The following are required only for core and extension modules (they are welcomed, but not required, for unsupported modules):

- [X] There is an issue in [Jira](https://osgeo-org.atlassian.net/projects/GEOT) describing the bug/task/new feature (a notable exemptions is, changes not visible to end users). The ticket is for the GeoTools project, if the issue was found elsewhere it's a good practice to link to the origin ticket/issue.
- [X] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [X] PR for bug fixes and small new features are presented as a single commit
- [X] Commit message(s) must be in the form "[GEOT-XYZW] Title of the Jira ticket"
- [x] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by Continuous Integration after opening this PR)
- [x] This PR passes the [QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html) (QA checks results will be reported by Continuous Integration after opening this PR)
- [ ] Documentation has been updated accordingly.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or not applicable.

### CODE REVIEW

File: modules/extension/tile-client/src/main/java/org/geotools/tile/Tile.java [hunk 1]
```diff
@@ -171,8 +181,17 @@ public BufferedImage getBufferedImage() {
         }
     }
 
+    /**
+     * Implementation of ImageLoader. Has been moved to {@link
+     * org.geotools.tile.TileService#loadImageTileImage(Tile)}
+     */
     public BufferedImage loadImageTileImage(Tile tile) throws IOException {
-        return ImageIOExt.readBufferedImage(getUrl());
+        final HTTPResponse response = service.getHttpClient().get(tile.getUrl());
```
[the remark this answers is not in the document]
Comment (roarbra) [reply modules/extension/tile-client/src/main/java/org/geotools/tile/Tile.java:192]: I saw that the three implementations had a reference to TileService, and tried to align it. There are more work to be done, but not in this PR.

**Comment by aaime** [comment]
@roarbra I was about to merge, when I realize I did not remember seeing a CLA from you on file.
I've double checked and could not find one from either you, or your organization (an online search seems to suggest [you work for NIVA](https://www.niva.no/en/Employees/roar-braenden)).

Yet, the checklist at the top of the PR has the "CLA" bit checked, so maybe you've sent one recently, and it's not yet on file? Can you clarify?

**Comment by roarbra** [comment]
> @roarbra I was about to merge, when I realize I did not remember seeing a CLA from you on file.
> I've double checked and could not find one from either you, or your organization (an online search seems to suggest [you work for NIVA](https://www.niva.no/en/Employees/roar-braenden)).
> 
> Yet, the checklist at the top of the PR has the "CLA" bit checked, so maybe you've sent one recently, and it's not yet on file? Can you clarify?

I was quite sure I sent that CLA, but something must have gone wrong. I sent the copy from 2018 now, but I could make a new one if that's necessary.

**Comment by aaime** [comment]
Roger that, got it. Merging this one.

**Comment by roarbra** [comment]
@aaime Would it be possible to back port this bug fix to the 23 and 24 versions?

**Pull request merged**